aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -41,6 +41,9 @@ class SchemaRegistrationHandleRAII;
  */
 class CAFFE2_API Dispatcher final {
 private:
+  // For direct access to backend fallback information
+  friend class impl::OperatorEntry;
```

Comment:
hm. This takes control away from the Dispatcher class and makes reasoning about thread-safety and mutexes harder. Could we instead pass down the full list of fallback kernels to OperatorEntry each time we register a kernel to it?